### PR TITLE
ShotoverProcessBuilder: fix with_profile

### PR DESCRIPTION
**Pull request by rukai**
blocked on: https://github.com/shotover/tokio-bin-process/pull/22

`with_profile` should overwrite `with_bin`.
`with_bin` is a hint to `ShotoverProcessBuilder` that there is a precompiled binary available.
`with_profile` is demanding that we use a specific profile which has priority over `with_bin`.

This bug was causing windsock to profile shotover binaries not suitable for profiling.

### CODE REVIEW

**Comment by rukai** [comment]
will need to investigate this failure

File: test-helpers/src/shotover_process.rs [hunk 1]
```diff
@@ -30,21 +30,27 @@ impl ShotoverProcessBuilder {
...
         self
     }
 
+    /// Run shotover with the specified number of cores
     pub fn with_cores(mut self, cores: u32) -> Self {
         self.cores = Some(cores.to_string());
         self
     }
 
+    /// Force shotover to recompiled with the specified profile
```
Comment (rukai) [inline test-helpers/src/shotover_process.rs:53]: `to be recompiled`

**Comment by rukai** [comment]
The mysterious GA failures were resolved by making tokio-bin-process only compile the exact binary it needs rather than the entire workspace: https://github.com/shotover/tokio-bin-process/pull/22

**Review by conorbros — APPROVED** [review]

**Comment by rukai** [comment]
Nix version needed to be updated for compatibility with the new tokio-bin-process version.

**Review by cjrolo — APPROVED** [review]
Thanks for the comments!

**Pull request merged**